File: leetcode/js/[342] Power of Four.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPowerOfFour = function(num) {
    if(num === 1) return true;
    
    let power = 4; 
    
    while(power <= num) {
        if(num === power) {
            return true;
        }
        power *= 4;
    }
    
    return false;
};

File: [342] Power of Four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n == 1:
          return True
        if n == 0 or n % 4 != 0:
          return False
        
        num = 1
        
        while num < n:
          num *= 4
          
        return num == n
        
        
